app/api/helpers/files.py

```diff
@@ -173,11 +163,11 @@ def create_save_image_sizes(image_file, image_sizes_type, unique_identifier=None
     return new_images
 
 
-def make_fe_url(settings=None, path=None):
+def make_frontend_url(path=None, parameters=None):
     """
     Create URL for frontend
     """
-    if settings is None:
```

Comment:
Build the url properly. Such that even if there are things like extra slashes or missing scheme in the frontend_url settings, the final url is proper. Use something like 

```python
frontend_url = urlparse.urlparse(settings['frontend_url'] if settings['frontend_url'] else '')
return urlparse.urlunparse((
		frontend_url.scheme,
		frontend_url.netloc,
		path,
		'',
		urllib.urlencode(parameters) if parameters else '', 
		''
	));
```